core/src/validation.jl

```diff
@@ -23,6 +23,28 @@ isnode(sv::Type{SchemaVersion{T, N}}) where {T, N} = length(split(string(T), "."
 nodetype(sv::Type{SchemaVersion{T, N}}) where {T, N} = Symbol.(split(string(T), ".")[2:3])
 nodetype(sv::SchemaVersion{T, N}) where {T, N} = Symbol.(split(string(T), ".")[2:3])
 
+# Allowed types for downstream (to_node_id) nodes given the type of the upstream (from_node_id) node
+neighbortypes(nodetype::Symbol) = neighbortypes(Val(nodetype))
+neighbortypes(::Val{:pump}) = Set((:basin,))
+neighbortypes(::Val{:basin}) = Set((
+    :linearresistance,
```

Comment:
So ideally we change the namespace of the validation as well? Because these are based on it, aligning with the nodetypes used also for the jsonschema.

Reply:
Yes exactly, that'd be great.

Reply:
Well namespaced can't be changed (Legolas rules), but I've changed all the Val{:type} to be camelcased like everywhere else.